### ISSUE TITLE
[codelabs] Breaking changes in Adaptive Apps in Flutter codelabs when using latest go_router

### ISSUE THREAD

**Issue by joshuadeguzman** [flutter/flutter#140948]
**Issue**

Currently, the examples shown in https://codelabs.developers.google.com/codelabs/flutter-adaptive-app#3 when adding go_router using the `flutter pub add` command will attempt to use the latest version from pub. This would cause issues for some new comers to Flutter not being able to run the sample code because of an outdated go_router API:

```
final title = state.queryParams['title']!;
final id = state.params['id']!;
```

`queryParams` and `params` have since changed pass go_router v7.

**Proposal**

The docs should change the reference to the deprecated method with the new APIs (nested for queries):

```
final title = state.uri.queryParameters['title']!;
final id = state.pathParameters['id']!;
```

Thank you.

**Comment by joshuadeguzman** [comment]
@domesticmouse happy to help changing any part of the codelabs or docs related to this, good write-up!

**Comment by domesticmouse** [comment]
Can you review the code in https://github.com/flutter/codelabs/tree/main/adaptive_app/step_07 and put together changes as a PR? I'll deal with back porting and updating the codelab text.

**Comment by joshuadeguzman** [comment]
> Can you review the code in https://github.com/flutter/codelabs/tree/main/adaptive_app/step_07 and put together changes as a PR? I'll deal with back porting and updating the codelab text.

Gotcha, I can put up a PR after my talk this afternoon. Thanks for referencing the project here!

**Comment by joshuadeguzman** [comment]
Hey @domesticmouse, upon checking I think there's not much work left here other than migrating one of the codelabs: boring_to_beautiful.

Perhaps this just warrants an update to the codelabs itself?

`state.params`
<img width="971" alt="Screenshot 2024-01-04 at 5 08 41 PM" src="https://github.com/flutter/flutter/assets/20706361/3d70e6b9-0672-45b6-881a-ec21cbaac2df">

`state.queryParams` 
<img width="687" alt="Screenshot 2024-01-04 at 5 08 59 PM" src="https://github.com/flutter/flutter/assets/20706361/35075f90-b375-4350-b4b7-bfda28e09025">

I'm fine closing this issue.

**Comment by domesticmouse** [comment]
I'll leave this open until I've updated the codelab text. It's out of date and pointing at the wrong repo. Thanks for the heads up!

**Comment by domesticmouse** [comment]
Googlers: fix in cl/595942881

**Comment by domesticmouse** [comment]
Landed back in January. Oops.
